<!doctype html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<title>全局变量的对比2</title>
</head>
<body>  
<script>  
   function test2(){  
       alert ("before for scope:"+i);    // i未赋值（并不是未声明！使用未声明的变量或函数全抛出致命错误而中断脚本执行）  
     
                                                       // 此时i的值是underfined  
       for(var i=0;i<3;i++){  
           alert("in for scope:"+i);  // i的值是 0、1、2, 当i为3时跳出循环  
       }  
       alert("after for scope:"+i);  // i的值是3，注意，此时已经在for scope以外，但i的值仍然保留为3  
         
       while(true){  
           var j = 1;  
           break;  
       }  
       alert(j);    // j的值是1，注意，此时已经在while scope以外，但j的值仍然保留为1  
     
       if(true){  
           var k = 1;  
       }  
       alert(k);  //k的值是1，注意，此时已经在if scope以外，但k的值仍然保留为1  
   }  
     
   test2();  
   //若在此时（function scope之外）再输出只存在于test2 这个function scope里的 i、j、k变量会发生神马效果呢？  
   alert(i); //error! 没错，是error，原因是变量i未声明（并不是未赋值，区分test2函数的第一行输出），导致脚本错误，程序到此结束！  
   alert("这行打印还会输出吗？"); //未执行  
   alert(j); //未执行  
   alert(k); //未执行  
</script>
</body>
</html>
<script>  
    var a =1;  
    function test(){     
        alert(window.a);  //a为1,这里的a是全局变量哦！  
        var a=2;     //局部变量a在这行定义  
        alert(a);  //a为2,这里的a是局部变量哦！  
    }  
    test();  
    alert(a); //a为1，这里并不在function scope内，a的值为全局变量的值  
</script>  